docs/en/observability/monitor-k8s/monitor-k8s-application-performance.asciidoc

```diff
@@ -0,0 +1,219 @@
+[discrete]
+[[monitor-kubernetes-application-performance]]
+== Part 3: Monitor application performance
+
+Quickly triage and troubleshoot application performance problems with the help of Elastic
+application performance monitoring (APM).
+
+Think of a latency spike -- APM can help you narrow the scope of your investigation to a single service.
+Because you've also ingested and correlated logs and metrics, you can then link the problem to CPU and memory utilization via the error log entries of a particular Kubernetes pod.
+
+[discrete]
+=== Step 1: Set up APM Server
+
+Application monitoring data is streamed from your applications running in Kubernetes to APM Server,
+where it is validated, processed, and transformed into {es} documents.
+
+There are many ways to deploy APM Server when working with Kubernetes,
+but this guide assumes that you're using our hosted {ess} on {ecloud}.
+If you haven't done so already, enable APM Server in the {ess-console}[{ess} console].
+
+If you want to manage APM Server yourself, there are a few alternative options:
+
+[%collapsible]
+.Expand alternatives
+====
+* {eck-ref}/[Elastic Cloud on Kubernetes (ECK)] -- The Elastic recommended approach for managing
+APM Server deployed with Kubernetes.
+Built on the Kubernetes Operator pattern, ECK extends basic Kubernetes orchestration capabilities
+to support the setup and management of APM Server on Kubernetes.
+
+* Deploy APM Server as a DaemonSet -- Ensure a running instance of APM Server on each node in your cluster.
+Useful when all pods in a node should share a single APM Server instance.
+
+* Deploy APM Server as a sidecar -- For environments that should not share an APM Server,
+like when directing traces from multiple applications to separate {es} clusters.
+
+* {apm-server-ref-v}/installing.html[Download and install APM Server] -- The classic, non-Kubernetes option.
+====
+
+[discrete]
+=== Step 2: Save your secret token
+
+A {apm-server-ref-v}/secret-token.html[secret token] is used to secure communication between APM agents
+and APM Server. On the {ecloud} deployment page, select *APM* and copy your APM Server secret token.
+If you're managing APM Server yourself,
+see {apm-server-ref-v}/secret-token.html[secret token] for instructions on how to set up your secret token.
+
+To avoid exposing the secret token, you can store it in a Kubernetes secret. For example:
+
+[source, cmd]
+----
+kubectl create secret generic apm-secret --from-literal=ELASTIC_APM_SECRET_TOKEN=asecretpassword --namespace=kube-system <1>
+----
+<1> Create the secret in the same namespace that you'll be deploying your applications in.
+
```

Comment:
```suggestion
If you are using ECK to set up APM Server, the operator will automatically create the `apm-server-apm-token` secret for you.

```
To be verified by someone from @elastic/apm-server team

Reply:
This looks to be correct: https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-apm-connecting.html#k8s-apm-secret-token.